### ISSUE TITLE
Spectral Linter: Provided ruleset is not an object

### ISSUE THREAD

**Issue by mprinold** [dweber019/backstage-plugins#82]
When implementing the Spectral Linter plugin i am finding that it does not want to accept the rulesets, i am using the example rules files you use in you dev examples:

config:
```
spectralLinter:
  openApiRulesetUrl: 'https://gist.githubusercontent.com/dweber019/a368819668a76363849db6378792e907/raw/backstage-spectral-linter-openapi-ruleset.yaml'
```

only error message is presented in the ui
<img width="1545" alt="Screenshot 2024-12-17 at 16 10 25" src="https://github.com/user-attachments/assets/95b6d6ce-dfa8-4e5d-ae31-6f664361cbb9" />

**Comment by Aaron-Mackay** [comment]
Hi, I've also been having this problem, though I've been using "spectral:oas" for that URL. The problem seems to arise from `this.ruleset = ruleset instanceof ruleset_1.Ruleset ? ruleset : new ruleset_1.Ruleset(ruleset);` in `node_modules/@dweber019/backstage-plugin-api-docs-spectral-linter/node_modules/@stoplight/spectral-core/dist/spectral.js`, as it seems to return negative when checking the instanceof, and then fail the assertion in the constructor of the Ruleset (I think because it's already gone through the process?) 

I can't quite work out why it's returning false, but will keep investigating and post here if I find a solution.

**Comment by Aaron-Mackay** [comment]
Looks like the reason is because of the origin of the `ruleset_1` import, `instanceof` sees them as different if the classes are from separate places. Not 100% sure on a proper fix (changing the import to `const ruleset_1 = require("../../../../../../@stoplight/spectral-core");` works, but doesn't feel quite right).

**Comment by dweber019** [comment]
Thanks @Aaron-Mackay came to same conclusion as you. Could you create a issue at spotlight for this problem?

**Comment by Aaron-Mackay** [comment]
Hi @dweber019 , I've been looking at it some more this morning, and found a possible different root.

It appears that normally, under `node_modules/@dweber019/backstage-plugin-api-docs-spectral-linter/node_modules/@stoplight`, there are only `spectral-core` and `types`. If you manually copy in `spectral-ruleset-bundler` from the root `node_modules`, everything works fine.

Unfortunately I don't have the expertise to understand why npm doesn't install the bundler locally to the plug-in - is there something that needs to be changed in the plugin's package.json?

**Comment by dweber019** [comment]
I created a patch / workaround in version @dweber019/backstage-plugin-api-docs-spectral-linter@0.4.10 for the plugin.
This is only a temporary fix until the upstream project has fixed it.

**Comment by anchitsingla** [comment]
It is not working for me still. Is it fixed ?

**Comment by anchitsingla** [comment]
@dweber019 I have tried it but still getting the same error

**Comment by lambav** [comment]
@dweber019 This isn’t working. Any update on this ?

**Comment by dweber019** [comment]
Yes, working on it. With the latest fix, at least in the Backstage instance I maintain in my company, it works.
Are you able to reproduce the issue with this repo or do you have me some logs, errors.

**Comment by dkadirbekovbeeline** [comment]
I have same problem with ^0.4.11 version. No logs in browser console. Spectral file is fetched successfully.

**Comment by Dutta805** [comment]
@Aaron-Mackay @dweber019 I am unable to locate the @stoplight folder under node modules of this plugin
node_modules/@dweber019/backstage-plugin-api-docs-spectral-linter/node_modules/@stoplight <-- not present

![Image](https://github.com/user-attachments/assets/8af983dc-fea0-47f6-9b35-0e48bf912783)


However @stoplight is present in the root of my node_modules, might this be the issue for the error - Provided ruleset is not an object?
in spectral.js -> const ruleset_1 = require("./ruleset/ruleset");

**Comment by benjidotsh** [comment]
Currently running into the same issue, although the network tab shows that the config is being successfully retrieved...

**Comment by benjidotsh** [comment]
@hspens I think you have found the culprit! I added `@stoplight/spectral-core@^1.20.0` as a resolution and it works fine now! Thanks!

I looked into the code of the plugin and I noticed the version of `@stoplight/spectral-core` is set to `1.19.5`. Perhaps the real fix would be to bump this to 1.20.0?

**Comment by dweber019** [comment]
Nice catch. I created a PR https://github.com/dweber019/backstage-plugins/pull/106 and released it with version 0.5.1

Could you test if it works for you.

**Comment by DawMatt** [comment]
@dweber019 , I experienced this same issue this morning.

Claude helped me track down the issue. In my case it ended up being two overlapping issues. In my instance this was solved by:
```
  "resolutions": {
    "@stoplight/spectral-ruleset-bundler": "1.7.0",
    "@stoplight/spectral-core": "1.23.0"
  }
```

Claude's explanation:

> ... breaks the Spectral tab (Failed to lint API / Provided ruleset is not an object), for
> two independent reasons:
> 
> 1. @stoplight/spectral-ruleset-bundler — the plugin pins 1.6.3, which resolves
> un-bundled modules through the now-defunct skypack CDN (cdn.skypack.dev) and fails in
> the browser. 1.7.0 switched to esm.sh and works (and is already pulled in by
> @dawmatt/api-grade-core).
> 2. @stoplight/spectral-core — the plugin pins 1.20.0, but the bundler and
> @dawmatt/api-grade-core use 1.23.0. The linter builds the ruleset with the bundler's
> spectral-core (1.23.0) and then hands it to new Spectral() from its own spectral-core
> (1.20.0). Spectral.setRuleset() does ruleset instanceof Ruleset ? … : new Ruleset(ruleset)
> — a 1.23.0 Ruleset is not instanceof the 1.20.0 Ruleset class, so it re-wraps a
> class instance as if it were a plain object and throws "Provided ruleset is not an object."
> Both must be the same spectral-core copy for the instanceof check to pass.

Item 1 impacts my test ruleset which simply extends the default Spectral rulesets. 
```
extends:
  - spectral:oas
  - spectral:asyncapi
```

Your existing fix is probably sufficient to deal with item 2 for most users. I had to increase the level purely due to other packages I'm using.
